### PR TITLE
e2e: grpc-gateway cURL tests

### PR DESCRIPTION
**Pull request by gyuho**
/cc @heyitsanthony @xiang90

### CODE REVIEW

File: e2e/v3_curl_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+func TestV3CurlPutGetNoTLS(t *testing.T)     { testCurlPutGetGRPCGateway(t, &configNoTLS) }
+func TestV3CurlPutGetAutoTLS(t *testing.T)   { testCurlPutGetGRPCGateway(t, &configAutoTLS) }
+func TestV3CurlPutGetAllTLS(t *testing.T)    { testCurlPutGetGRPCGateway(t, &configTLS) }
+func TestV3CurlPutGetPeerTLS(t *testing.T)   { testCurlPutGetGRPCGateway(t, &configPeerTLS) }
+func TestV3CurlPutGetClientTLS(t *testing.T) { testCurlPutGetGRPCGateway(t, &configClientTLS) }
+func testCurlPutGetGRPCGateway(t *testing.T, cfg *etcdProcessClusterConfig) {
+	defer testutil.AfterTest(t)
+
+	// test doesn't use quorum gets, so ensure there are no followers to avoid
```
Comment (heyitsanthony) [inline e2e/v3_curl_test.go]: Is this correct? I think the Ranges being submitted are serialized by default

Comment (gyuho) [inline e2e/v3_curl_test.go]: You are right. Will fix.

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
@xiang90 As you suggested, I changed it just to use json package to encode structs. PTAL.

**Comment by xiang90** [comment]
lgtm when fix the CI issues.

**Comment by gyuho** [comment]
thanks. merging in greens.

**Pull request merged**